extension-client/src/util/workspace.ts

```diff
@@ -656,3 +656,24 @@ export function readUIAssetManifest(path: string): any {
     });
 }
 
+export async function getResourceGroupInput(): Promise<string> {
+         // Get the Resource Group
+         const inputResourceGroup = await vscode.window.showInputBox({
+            ignoreFocusOut: true,
+            placeHolder: 'Type resource group name, press escape to use default resource group',
+            prompt: 'Resource group where you want to organize your workspace. ',
+        });
+        return new Promise(async (resolve, reject) => {
+                if (inputResourceGroup) {
+                    // user entered a string value
+                    resolve(inputResourceGroup);
+                } else if (inputResourceGroup === undefined){
+                    // user pressed ESC
+                    resolve('Default');
+                } else {
+                    // any other error
+                    reject('Resource Group rejected!');
+                }
```

Comment:
What could be any other errors here? 
I see user can: 
- Type in a string and press enter
- Type in empty spaces and press enter or Just press enter
- Type nothing and press escape button

In either of the above cases, logic is if user enters some valid string set that as resource group or fallback to `Default`

Check above scenarios and decide if there is a need to have `reject('Resource Group rejected!');` 

As I was running the code I see when user just presses enter without entering anything, deploy fails.